src/accelerate/accelerator.py

```diff
@@ -1655,19 +1656,14 @@ def _prepare_ipex(self, *args):
             elif isinstance(obj, (torch.optim.Optimizer)):
                 optimizer = obj
         if optimizer is not None and model is not None:
-            if is_ipex_available():
-                if is_xpu_available() and self.device.type == "xpu":
-                    model = model.to(self.device)
-                    model, optimizer = torch.xpu.optimize(model, optimizer=optimizer, inplace=True, level="O1")
-                    model.forward = torch.xpu.amp.autocast()(model.forward)
-                else:
-                    model, optimizer = ipex.optimize(model, optimizer=optimizer, inplace=True, level="O1")
-                    model.forward = torch.cpu.amp.autocast()(model.forward)
+            dtype = torch.bfloat16 if self.state.mixed_precision == "bf16" else torch.float32
+            if is_xpu_available() and self.device.type == "xpu":
+                model = model.to(self.device)
+                model, optimizer = torch.xpu.optimize(
+                    model, optimizer=optimizer, dtype=dtype, inplace=True, level="O1"
+                )
             else:
-                if is_torch_version(">=", "1.10"):
-                    model.forward = torch.autocast(self.device.type)(model.forward)
-                else:
-                    model.forward = convert_outputs_to_fp32(model.forward)
+                model, optimizer = ipex.optimize(model, optimizer=optimizer, dtype=dtype, inplace=True, level="O1")
```

Comment:
We should definitely not be doing this, the native mixed precision autocast logic needs to stay when not using ipex. 

This second check for `ipex.optimize` should be under the xpu available check only or specific to `ipex`.

Reply:
hi, @muellerzr , there's logic in https://github.com/huggingface/accelerate/blob/main/src/accelerate/accelerator.py#L1307. after prepare_ipex. it will go there for the amp disposal.

Reply:
ipex.optimize does not have device type input, it is for cpu now. for xpu, torch.xpu.optimize is used

Reply:
Got it, that makes sense there then now, excellent. Sorry for the confusion!